packages/vm/src/evm/evm.ts

```diff
@@ -407,12 +408,26 @@ export default class EVM {
         if (!this._vm._common.isActivatedEIP(3540)) {
           result = { ...result, ...INVALID_BYTECODE_RESULT(message.gasLimit) }
         }
-        // EIP-3540 EOF1 checks
-        if (!eof1CodeAnalysis(result.returnValue)) {
+        // EIP-3540 EOF1 header check
+        const eof1CodeAnalysisResults = eof1CodeAnalysis(result.returnValue)
+        if (!eof1CodeAnalysisResults?.code) {
           result = {
             ...result,
             ...INVALID_BYTECODE_RESULT(message.gasLimit),
           }
+        } else if (this._vm._common.isActivatedEIP(3670)) {
+          // EIP-3670 EOF1 code check
+          const codeStart = eof1CodeAnalysisResults.data > 0 ? 10 : 7
```

Comment:
Can you add an additional comment on the meaning of this 10, 7?

Reply:
Done.